server/@types/forms/index.d.ts

```diff
@@ -37,3 +37,23 @@ declare module 'forms' {
     status: 'NOT_STARTED' | 'ACTIVE' | 'COMPLETE'
   }
 }
+
+/**
+ * Module declaring "composite forms", specifically types that contain other forms.
+ * The composite form is not a form object in its own right, in that there are no HTML views, forms or express request
+ * handlers that bind request form fields to an object of this type. It is a convenience object to allow collating
+ * several form objects into a single object held in the session; typically used in multi-page / wizard style screens.
+ */
+declare module 'compositeForms' {
+  import type { AddNoteForm, AddStepForm, CreateGoalForm } from 'forms'
+
+  /**
+   * A composite form representing the individual form objects for creating a new Goal
+   */
+  export interface NewGoal {
```

Comment:
Whilst reading the description, I wondered if `NewGoalForms` would be better, but now I'm not so sure given it's already within the `compositeForms` module. Also, I guess that could become confusing when you introduce an array of these (if that's your intention).

Reply:
Yeah; naming is hard! and I'm defo open to a better name than `NewGoal` !

Even though its in the `compositeForms` module, it felt "wrong" to call it a "Form" because its not a form in its own right. IE. there are no html forms or request handlers that bind to the properties of this object. Its just a convenient container that contains other form objects.

And yes, naming it plural (`NewGoalForms`) would be really confusing when we start using it in an array! 😁 
Currently its a single instance on the session:
```
interface SessionData {
    newGoalForm: NewGoal
```
(hmmm - maybe the property name on the session is wrong, and it should just be `newGoal` ?

If we called the type `NewGoalForm` it would kinda work (but I dont like it because its not a form! 🤣 )
```
interface SessionData {
    newGoalForm: NewGoalForm
```

If we called the type `NewGoalForms` it would start looking wrong ....
```
interface SessionData {
    newGoalForm: NewGoalForms
```

And then once it becomes an array ....
```
interface SessionData {
    newGoalForms: Array<NewGoalForms>
```

Naming is hard! 🤣